### ISSUE TITLE
Is this really RFC4180 compliant?

### ISSUE THREAD

**Issue by dooferorg** [JoshClose/CsvHelper#2022]
Trying to parse a CSV file that is tab delimited and has all fields quoted. I get a 'bad data' error with it, despite having the mode set to RFC4180.

RFC4180 states 'Each field may or may not be enclosed in double quotes' (https://www.rfc-editor.org/rfc/rfc4180.html) .. So my question is why is the file not being parsed correctly?

Excel can read it fine setting delimiter to tab and 'text qualifier' to a double quote (in the legacy wizards text importer)

Using CSVHelper 28.0.1 via nuget in a C# .net 6 program.

Config:

```
CsvConfiguration configuration = new CsvConfiguration(System.Globalization.CultureInfo.CurrentCulture)
                {
                    Mode = CsvMode.RFC4180,
                    Delimiter = "\t",
                    HasHeaderRecord = true,
                    IgnoreBlankLines = true,
                    LineBreakInQuotedFieldIsBadData = false,
                    TrimOptions = CsvHelper.Configuration.TrimOptions.Trim
                };
```

Not sure how to deal with that but it seems like it should be possible to configure it correctly to handle the file, I just don't know how to get there. However, I would have thought that if this was truely RGC4180 compliant then it should handle this file automatically.

**Comment by AltruCoder** [comment]
My guess is that you have an un-escaped double quote inside your field.  RFC4180 also states:
> If double-quotes are used to enclose fields, then a double-quote appearing inside a field must be escaped by preceding it with another double quote.

* The first one with tabs and quoted fields works.   `"1"  "MyName"`
* The second one that has an un-escaped double quote in the name fails as bad data.  `"1"  "My"Name"`
* Third one works if the double quote in MyName is escaped with another double quote `"1"  "My""Name"`
```csharp
void Main()
{
	var config = new CsvConfiguration(CultureInfo.InvariantCulture)
	{
		Mode = CsvMode.RFC4180,
		Delimiter = "\t",
		HasHeaderRecord = true,
		IgnoreBlankLines = true,
		LineBreakInQuotedFieldIsBadData = false,
		TrimOptions = CsvHelper.Configuration.TrimOptions.Trim
	};

	using (var reader = new StringReader("\"Id\"\t\"Name\"\n\"1\"\t\"MyName\""))
	using (var csv = new CsvReader(reader, config))
	{
		var records = csv.GetRecords<Foo>().Dump();
	}


	using (var reader = new StringReader("\"Id\"\t\"Name\"\n\"1\"\t\"My\"Name\""))
	using (var csv = new CsvReader(reader, config))
	{
		try
		{	        
			var records = csv.GetRecords<Foo>().Dump();
		}
		catch (Exception ex)
		{			
			ex.Dump();
		}		
	}

	using (var reader = new StringReader("\"Id\"\t\"Name\"\n\"1\"\t\"My\"\"Name\""))
	using (var csv = new CsvReader(reader, config))
	{
		var records = csv.GetRecords<Foo>().Dump();
	}
}

public class Foo
{
	public int Id { get; set; }
	public string Name { get; set; }
}
```

**Comment by dooferorg** [comment]
Well, no, it's blowing out on the headers even and they're fine. As I said, this file opens fine in Excel.

**Comment by AltruCoder** [comment]
Can you provide some sample data that doesn't work?

**Comment by dooferorg** [comment]
I'll give it a go to come up with a sanitized subset and report. The data includes customer data so I need to be careful what I'm sharing.

**Comment by dooferorg** [comment]
So, you're all quite right about the bad data! I WAS able to find the offending row (checking ReadingContext for the RawRow value) and had misinterpreted the exception message I saw with version 28 thinking it was complaining about the headers when it was not that. The exception looks slightly different with version 18, which I was using and have gone back to.

I had moved the export of the information in question to not requiring any sort of text qualifier on its export (so just using tab delimiting) and haven't had any problems. The export from the company I was getting the data from did indeed not escape the double quote character properly in their exports.

So, sorry for my misunderstanding this. I'm glad to be wrong though that it's doing things properly. You're likely also right that Excel fixed this munged data so it shouldn't be relied on for comparing parsing of files.
